packages/fxa-settings/src/models/integrations/oauth-integration.ts

```diff
@@ -197,6 +198,43 @@ export class OAuthIntegration extends BaseIntegration<OAuthIntegrationFeatures>
     return this.data.redirectUri;
   }
 
+  getRedirectToRPUrl(
+    extraParams: Record<string, number | string | undefined> = {}
+  ) {
+    const redirectUrl = new URL(this.getRedirectUri());
+    const params = {
+      ...extraParams,
+
+      action: this.data.action,
+      state: this.data.state,
+
+      flowId: this.data.flowId,
+      flowBeginTime: this.data.flowBeginTime,
+      deviceId: this.data.deviceId,
+
+      utmCampaign: this.data.utmCampaign,
+      utmContent: this.data.utmContent,
+      utmMedium: this.data.utmMedium,
+      utmSource: this.data.utmSource,
+      utmTerm: this.data.utmTerm,
+    };
+
+    const redirectParams = Object.fromEntries(
+      Object.entries(params).filter(([_, v]) => v !== undefined)
+    );
+
+    redirectUrl.search = new URLSearchParams(redirectParams).toString();
+    return redirectUrl.toString();
+  }
+
+  getRedirectWithErrorUrl(err: AuthUiError) {
+    return this.getRedirectToRPUrl({ error: err.errno });
+  }
+
+  getRedirectWithOAuthResult() {
+    return this.getRedirectToRPUrl();
+  }
```

Comment:
These redirect related functions probably will go away once the OAuth signin work provides alternative(s).  Or at least `getRedirectWithOAuthResult` will get an actual implementation that includes the authorization code then.